app/javascript/features/form/components/DialogContainer.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  Button
+} from "@material-ui/core";
+import React from "react";
+
+type OpenDialog = {
+  isOpenSuccessDialog: boolean;
+  setIsOpenSuccessDialog: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
```suggestion
  onClose: () => void
```
この引数にはダイアログを閉じたいときに呼び出す関数が入るから、名前はonCloseにして型は() => voidでどうでしょう